docs/client.rst

```diff
@@ -111,12 +111,33 @@ For incorporating the changes to your project, is necessary to run the ``cmake``
     This value corresponds to the `Maximun Transmission Unit` able to send and receive by Serial.
     Internally a buffer is created proportional to this size.
 
+Streams
+-------
+The client communication is performed by streams.
+The streams can be seen as communication channels.
+There are two types of streams: best effort and reliable streams and you can create several of them.
+
+- Best effort streams will send and receive the data leaving the reliability to the transport layer.
+  As a result, the best effort streams consume fewer resources than a reliable stream.
+
+- Reliable streams perform the communication without lost regardless of the transport layer.
+  To avoid message losses, the reliable streams use additional messages to confirm the delivery, along to a history of the messages sent and received.
+  The history is used to storage messages that can no be currently processed because of the delivery order, or must be send again if the message can not be confirmed.
+  If the history is full, all messages written to the agent or received from the agent will be discarded, and must be requested later.
+  For that, a low history causes that more messages to be discarted, increasing the data traffic beacusen they need to be sent again.
```

Comment:
```suggestion
  For that, a short history causes that more messages to be discarded, increasing the data traffic because they need to be sent again.
```